sia/host/host.go

```diff
@@ -0,0 +1,105 @@
+package host
+
+import (
+	"errors"
+	"io"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/hash"
+	"github.com/NebulousLabs/Sia/sia/components"
+)
+
+const (
+	StorageProofReorgDepth = 6 // How many blocks to wait before submitting a storage proof.
+	maxContractLen         = 1 << 24
+)
+
+type Host struct {
+	announcement   components.HostAnnouncement
+	spaceRemaining int64
+	wallet         components.Wallet
+	state          components.ReadOnlyState
+
+	height          consensus.BlockHeight      // Current height of the state.
+	transactionChan chan consensus.Transaction // Can send channels to the state.
+
+	hostDir     string
+	fileCounter int
+	files       map[hash.Hash]string
+
+	forwardContracts  map[consensus.BlockHeight][]ContractEntry
+	backwardContracts map[consensus.BlockHeight][]ContractEntry
+
+	rwLock sync.RWMutex
```

Comment:
OH I get it. It's so that the locks aren't exported.

I guess it's safer this way. But having a whole `lock.go` file in all these packages really bothers me.